### PR TITLE
add temperature dependence to MSMR model

### PR DESCRIPTION
**Pull request by valentinsulzer**
# Description

The FR/T term now uses the appropriate local temperature. Individual parameters (x_j, U0_j, w_j, alpha_j) can also depend on temperature, but this isn't tested. Plett Volume 3 claims that those parameters should not change much as a function of temperature.

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/models/submodels/particle/msmr_diffusion.py [hunk 1]
```diff
@@ -143,24 +124,11 @@ def get_fundamental_variables(self):
             if self.x_average is True:
                 U = pybamm.SecondaryBroadcast(U, [f"{domain} electrode"])
 
+        # Accessing the actual cell temperature requires more refactoring since
```
Comment (rtimms) [inline pybamm/models/submodels/particle/msmr_diffusion.py:127]: incomplete comment

**Review by rtimms — COMMENTED** [review]
Thanks, looks good, just a couple of things to tidy up.

We should make it clear in the docs/example how `dUdT` is used in the MSMR model. In the "standard" model `U(x,T) = U_ref(x) + (T-T_ref)*dUdT` whereas here `U(x,T)` is solved for explicitly, and `dUdT` is just used to calculate the reversible heat term.

File: CHANGELOG.md [hunk 2]
```diff
@@ -33,14 +34,15 @@
 - Added a method, `insert_reference_electrode`, to `pybamm.lithium_ion.BaseModel` that insert a reference electrode to measure the electrolyte potential at a given position in space and adds new variables that mimic a 3E cell setup. ([#3573](https://github.com/pybamm-team/PyBaMM/pull/3573))
 - Serialisation added so models can be written to/read from JSON ([#3397](https://github.com/pybamm-team/PyBaMM/pull/3397))
 - Added a `get_parameter_info` method for models and modified "print_parameter_info" functionality to extract all parameters and their type in a tabular and readable format ([#3584](https://github.com/pybamm-team/PyBaMM/pull/3584))
-- Mechanical parameters are now a function of stoichiometry and temperature ([#3576](https://github.com/pybamm-team/PyBaMM/pull/3576))
```
[the remark this answers is not in the document]
Comment (tinosulzer) [reply CHANGELOG.md:36]: yes, it's duplicate in the changelog

**Review by rtimms — APPROVED** [review]

**Comment by brosaplanella** [comment]
Happy to merge this one once the conflicts are resolved. There are also some coverage issues, but the detailed report does not load, so it might be a Codecov artifact.

**Pull request merged**